system/lib/pthread/emscripten_yield.c

```diff
@@ -0,0 +1,43 @@
+#include <stdbool.h>
+#include <sched.h>
+#include <threads.h>
+#include "syscall.h"
+
+#include <emscripten/threading.h>
+
+static _Atomic bool thread_crashed = false;
+
+void _emscripten_thread_crashed() {
+  thread_crashed = true;
+}
+
+/*
+ * Called whenever a thread performs a blocking action (or calls
+ * sched_yeild).
+ * This function takes care of running the event queue as well
+ * as keep the threads loaded code in sync (in case one thread
+ * calls dlopne()).
+ */
+void _emscripten_yield() {
+  int is_runtime_thread = emscripten_is_main_runtime_thread();
```

Comment:
is the concept of `main_browser_thread` vs `main_runtime_thread` still a thing? If this were the main runtime thread but running in a worker, would we still need to process queued calls? (or conversely, would secondary workers still want to process queued calls?)

Reply:
Yes.. the runtime thread is the thread on which the application was started.. it might be a worker.  The main browser thread is the one on which you can't block.   There is always a main runtime thread, but there could be no browser thread.

Reply:
Normally of course they are the same.

Also, if the main browser thread exists then it is always, by definition, the main runtime thread too.